src/fpga_mezz.c

```diff
@@ -0,0 +1,702 @@
+#include <stdio.h>
+#include <stdlib.h>
+
+/* mraa header */
+#include "mraa/i2c.h"
+
+#include "fpga_mezz.h"
+
+#define ARD_I2C_0_FUNC 0
+#define ARD_SPI_FUNC 1
+#define ARD_UART_FUNC 2
+#define ARD_I2C_1_FUNC 3
+
+#define I2C_BUS 0
+
+#define INPUT 1
+#define OUTPUT 0
+
+#define HIGH 1
+#define LOW 0
+
+#define FUNC_PASSTHROUGH 0
+#define FUNC_GPIO 1
+
+u_int8_t check_version(mraa_i2c_context i2c)
+{
+    u_int8_t ret;
+
+    ret = mraa_i2c_read_byte_data(i2c, REV_REG);
+    if (ret == -1) {
+        return EXIT_FAILURE;
+    }
+    
+    return ret;
+}
+
+int test(mraa_i2c_context i2c)
+{
+
+    u_int8_t ret;
+    mraa_result_t status = MRAA_SUCCESS;
+
+    status = mraa_i2c_write_byte_data(i2c, 0x80, TEST_REG);
+    if (status != MRAA_SUCCESS) {
+        printf("MRAA ERROR\n");
```

Comment:
No error return? Also, it makes sense to call `mraa_i2c_stop` for all failures.

Reply:
all i2c errors now call mraa_i2c_stop before return.